packages/web-app-files/src/mixins/actions/rename.ts

```diff
@@ -10,243 +10,243 @@ import {
+      !resources[0].isFolder && !areFileExtensionsShown ? nameWithoutExtension : resources[0].name
+
+    const title = $gettextInterpolate(
+      resources[0].isFolder ? $gettext('Rename folder %{name}') : $gettext('Rename file %{name}'),
+      { name: modalTitle }
+    )
+
+    const inputValue =
+      !resources[0].isFolder && !areFileExtensionsShown ? nameWithoutExtension : resources[0].name
+
+    const inputSelectionRange =
+      resources[0].isFolder || !areFileExtensionsShown ? null : [0, nameWithoutExtension.length]
+
+    const modal = {
+      variation: 'passive',
+      title,
+      cancelText: $gettext('Cancel'),
+      confirmText: $gettext('Rename'),
+      hasInput: true,
+      inputValue,
+      inputSelectionRange,
+      inputLabel: resources[0].isFolder ? $gettext('Folder name') : $gettext('File name'),
+      onCancel: () => store.dispatch('hideModal'),
+      onConfirm: confirmAction,
+      onInput: checkName
+    }
 
-    async $_rename_renameResource(resource: Resource, newName: string, space?: SpaceResource) {
-      this.toggleModalConfirmButton()
+    store.dispatch('createModal', modal)
+  }
 
-      try {
-        space = space || this.space
-        const newPath = join(dirname(resource.path), newName)
-        await (this.$clientService.webdav as WebDAV).moveFiles(space, resource, space, {
-          path: newPath
-        })
-        this.hideModal()
-
-        const isCurrentFolder = isSameResource(resource, this.currentFolder)
-
-        if (isShareSpaceResource(space) && resource.isReceivedShare()) {
-          space.rename(newName)
-
-          if (isCurrentFolder) {
-            const currentFolder = { ...this.currentFolder } as Resource
-            currentFolder.name = newName
-            this.SET_CURRENT_FOLDER(currentFolder)
-            return this.$router.push(
-              createFileRouteOptions(space, {
-                path: '',
-                fileId: resource.fileId
-              })
-            )
-          }
-
-          const sharedResource = { ...resource }
-          sharedResource.name = newName
-          this.UPSERT_RESOURCE(sharedResource)
-          return
+  const actions = computed((): Action[] => [
+    {
+      name: 'rename',
+      icon: 'pencil',
+      label: () => {
+        return $gettext('Rename')
+      },
+      handler,
+      isEnabled: ({ resources }) => {
+        if (isLocationTrashActive(router, 'files-trash-generic')) {
+          return false
         }
-
-        if (isCurrentFolder) {
-          const currentFolder = { ...this.currentFolder } as Resource
-          renameResource(space, currentFolder, newPath)
-          this.SET_CURRENT_FOLDER(currentFolder)
-          return this.$router.push(
-            createFileRouteOptions(this.space, {
-              path: newPath,
-              fileId: resource.fileId
-            })
-          )
+        if (isLocationSharesActive(router, 'files-shares-with-me') && !unref(canRename)) {
```

Comment:
You introduced `useCapabilityFilesSharingCanRename` which defaults to `false`. This is an issue with oC10, because it does not send the capability although it can rename shares -> that's why the pipeline fails.

I guess we can just default it to `true` 🤷